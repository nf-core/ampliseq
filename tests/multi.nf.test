nextflow_pipeline {

    name "Test pipeline"
    script "../main.nf"
    tag "pipeline"
    profile "test_multi"

    test("-profile test_multi") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])

            assert workflow.success
            assertAll(
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we test pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/software_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // Manually chosen files for content checks
                    path("$outputDir/overall_summary.tsv"),
                    path("$outputDir/barrnap/rrna.arc.gff"),
                    path("$outputDir/barrnap/rrna.bac.gff"),
                    path("$outputDir/barrnap/rrna.euk.gff"),
                    path("$outputDir/barrnap/rrna.mito.gff"),
                    path("$outputDir/dada2/ASV_seqs.fasta"),
                    path("$outputDir/dada2/ASV_table.tsv"),
                    path("$outputDir/dada2/ref_taxonomy.rdp_18.txt"),
                    path("$outputDir/dada2/DADA2_stats.tsv"),
                    path("$outputDir/dada2/DADA2_table.rds"),
                    path("$outputDir/dada2/DADA2_table.tsv"),
                    path("$outputDir/input/Samplesheet_multi.tsv"),
                    path("$outputDir/multiqc/multiqc_data/multiqc_general_stats.txt"),
                    path("$outputDir/multiqc/multiqc_data/multiqc_fastqc.txt"),
                    path("$outputDir/qiime2/representative_sequences/descriptive_stats.tsv"),
                    path("$outputDir/qiime2/representative_sequences/seven_number_summary.tsv")
                ).match() }
            )
        }
    }
}
