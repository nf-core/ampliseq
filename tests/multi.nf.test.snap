{
    "-profile test_multi": {
        "content": [
            39,
            {
                "BARRNAP": {
                    "barrnap": 0.9
                },
                "BARRNAPSUMMARY": {
                    "python": "Python 3.9.1"
                },
                "COMBINE_TABLE_DADA2": {
                    "R": "4.0.3"
                },
                "DADA2_DENOISING": {
                    "R": "4.3.2",
                    "dada2": "1.30.0"
                },
                "DADA2_ERR": {
                    "R": "4.3.2",
                    "dada2": "1.30.0"
                },
                "DADA2_FILTNTRIM": {
                    "R": "4.3.2",
                    "dada2": "1.30.0"
                },
                "DADA2_MERGE": {
                    "R": "4.1.1",
                    "dada2": "1.22.0"
                },
                "DADA2_RMCHIMERA": {
                    "R": "4.3.2",
                    "dada2": "1.30.0"
                },
                "DADA2_STATS": {
                    "R": "4.3.2",
                    "dada2": "1.30.0"
                },
                "DADA2_TAXONOMY": {
                    "R": "4.3.2",
                    "dada2": "1.30.0"
                },
                "FASTQC": {
                    "fastqc": "0.12.1"
                },
                "FILTER_STATS": {
                    "python": "3.9.1",
                    "pandas": "1.1.5"
                },
                "FORMAT_TAXONOMY": {
                    "bash": "5.0.16"
                },
                "MERGE_STATS_FILTERTAXA": {
                    "R": "4.3.2"
                },
                "QIIME2_BARPLOT": {
                    "qiime2": "2024.10.1"
                },
                "QIIME2_EXPORT_ABSOLUTE": {
                    "qiime2": "2024.10.1"
                },
                "QIIME2_EXPORT_RELASV": {
                    "qiime2": "2024.10.1"
                },
                "QIIME2_EXPORT_RELTAX": {
                    "qiime2": "2024.10.1"
                },
                "QIIME2_INASV": {
                    "qiime2": "2024.10.1"
                },
                "QIIME2_INSEQ": {
                    "qiime2": "2024.10.1"
                },
                "QIIME2_SEQFILTERTABLE": {
                    "qiime2": "2024.10.1"
                },
                "QIIME2_TABLEFILTERTAXA": {
                    "qiime2": "2024.10.1"
                },
                "RENAME_RAW_DATA_FILES": {
                    "sed": 4.7
                },
                "Workflow": {
                    "nf-core/ampliseq": "v2.15.0dev"
                }
            },
            [
                "barrnap",
                "barrnap/rrna.arc.gff",
                "barrnap/rrna.bac.gff",
                "barrnap/rrna.euk.gff",
                "barrnap/rrna.mito.gff",
                "barrnap/summary.tsv",
                "dada2",
                "dada2/ASV_seqs.fasta",
                "dada2/ASV_table.tsv",
                "dada2/ASV_tax.rdp_18.tsv",
                "dada2/DADA2_stats.tsv",
                "dada2/DADA2_table.rds",
                "dada2/DADA2_table.tsv",
                "dada2/QC",
                "dada2/QC/1_1.err.convergence.txt",
                "dada2/QC/1_1.err.pdf",
                "dada2/QC/1_2.err.convergence.txt",
                "dada2/QC/1_2.err.pdf",
                "dada2/QC/2_1.err.convergence.txt",
                "dada2/QC/2_1.err.pdf",
                "dada2/QC/2_2.err.convergence.txt",
                "dada2/QC/2_2.err.pdf",
                "dada2/QC/svg",
                "dada2/QC/svg/1_1.err.svg",
                "dada2/QC/svg/1_2.err.svg",
                "dada2/QC/svg/2_1.err.svg",
                "dada2/QC/svg/2_2.err.svg",
                "dada2/args",
                "dada2/args/assignTaxonomy.args.txt",
                "dada2/args/dada.args.txt",
                "dada2/args/filterAndTrim.args.txt",
                "dada2/args/learnErrors.args.txt",
                "dada2/args/mergePairs.args.txt",
                "dada2/args/removeBimeraDenovo.args.txt",
                "dada2/chunks",
                "dada2/chunks/ASV_seqs.1.ASV_tax.rdp_18.tsv",
                "dada2/log",
                "dada2/log/1.dada.log",
                "dada2/log/1.err.log",
                "dada2/log/2.dada.log",
                "dada2/log/2.err.log",
                "dada2/ref_taxonomy.rdp_18.txt",
                "fastqc",
                "fastqc/sampleID_1_1_fastqc.html",
                "fastqc/sampleID_1_2_fastqc.html",
                "fastqc/sampleID_1a_1_fastqc.html",
                "fastqc/sampleID_1a_2_fastqc.html",
                "fastqc/sampleID_2_1_fastqc.html",
                "fastqc/sampleID_2_2_fastqc.html",
                "fastqc/sampleID_2a_1_fastqc.html",
                "fastqc/sampleID_2a_2_fastqc.html",
                "input",
                "input/Samplesheet_multi.tsv",
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_data/fastqc-status-check-heatmap.txt",
                "multiqc/multiqc_data/fastqc_adapter_content_plot.txt",
                "multiqc/multiqc_data/fastqc_overrepresented_sequences_plot.txt",
                "multiqc/multiqc_data/fastqc_per_base_n_content_plot.txt",
                "multiqc/multiqc_data/fastqc_per_base_sequence_quality_plot.txt",
                "multiqc/multiqc_data/fastqc_per_sequence_gc_content_plot_Counts.txt",
                "multiqc/multiqc_data/fastqc_per_sequence_gc_content_plot_Percentages.txt",
                "multiqc/multiqc_data/fastqc_per_sequence_quality_scores_plot.txt",
                "multiqc/multiqc_data/fastqc_sequence_counts_plot.txt",
                "multiqc/multiqc_data/fastqc_sequence_duplication_levels_plot.txt",
                "multiqc/multiqc_data/fastqc_sequence_length_distribution_plot.txt",
                "multiqc/multiqc_data/fastqc_top_overrepresented_sequences_table.txt",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc_citations.txt",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_fastqc.txt",
                "multiqc/multiqc_data/multiqc_general_stats.txt",
                "multiqc/multiqc_data/multiqc_sources.txt",
                "multiqc/multiqc_plots",
                "multiqc/multiqc_plots/pdf",
                "multiqc/multiqc_plots/pdf/fastqc-status-check-heatmap.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_adapter_content_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_overrepresented_sequences_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_per_base_n_content_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_per_base_sequence_quality_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_per_sequence_gc_content_plot_Counts.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_per_sequence_gc_content_plot_Percentages.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_per_sequence_quality_scores_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_sequence_counts_plot-cnt.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_sequence_counts_plot-pct.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_sequence_duplication_levels_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_sequence_length_distribution_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_top_overrepresented_sequences_table.pdf",
                "multiqc/multiqc_plots/pdf/general_stats_table.pdf",
                "multiqc/multiqc_plots/png",
                "multiqc/multiqc_plots/png/fastqc-status-check-heatmap.png",
                "multiqc/multiqc_plots/png/fastqc_adapter_content_plot.png",
                "multiqc/multiqc_plots/png/fastqc_overrepresented_sequences_plot.png",
                "multiqc/multiqc_plots/png/fastqc_per_base_n_content_plot.png",
                "multiqc/multiqc_plots/png/fastqc_per_base_sequence_quality_plot.png",
                "multiqc/multiqc_plots/png/fastqc_per_sequence_gc_content_plot_Counts.png",
                "multiqc/multiqc_plots/png/fastqc_per_sequence_gc_content_plot_Percentages.png",
                "multiqc/multiqc_plots/png/fastqc_per_sequence_quality_scores_plot.png",
                "multiqc/multiqc_plots/png/fastqc_sequence_counts_plot-cnt.png",
                "multiqc/multiqc_plots/png/fastqc_sequence_counts_plot-pct.png",
                "multiqc/multiqc_plots/png/fastqc_sequence_duplication_levels_plot.png",
                "multiqc/multiqc_plots/png/fastqc_sequence_length_distribution_plot.png",
                "multiqc/multiqc_plots/png/fastqc_top_overrepresented_sequences_table.png",
                "multiqc/multiqc_plots/png/general_stats_table.png",
                "multiqc/multiqc_plots/svg",
                "multiqc/multiqc_plots/svg/fastqc-status-check-heatmap.svg",
                "multiqc/multiqc_plots/svg/fastqc_adapter_content_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc_overrepresented_sequences_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc_per_base_n_content_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc_per_base_sequence_quality_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc_per_sequence_gc_content_plot_Counts.svg",
                "multiqc/multiqc_plots/svg/fastqc_per_sequence_gc_content_plot_Percentages.svg",
                "multiqc/multiqc_plots/svg/fastqc_per_sequence_quality_scores_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc_sequence_counts_plot-cnt.svg",
                "multiqc/multiqc_plots/svg/fastqc_sequence_counts_plot-pct.svg",
                "multiqc/multiqc_plots/svg/fastqc_sequence_duplication_levels_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc_sequence_length_distribution_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc_top_overrepresented_sequences_table.svg",
                "multiqc/multiqc_plots/svg/general_stats_table.svg",
                "multiqc/multiqc_report.html",
                "overall_summary.tsv",
                "pipeline_info",
                "pipeline_info/software_versions.yml",
                "qiime2",
                "qiime2/abundance_tables",
                "qiime2/abundance_tables/abs-abund-table-2.tsv",
                "qiime2/abundance_tables/abs-abund-table-3.tsv",
                "qiime2/abundance_tables/abs-abund-table-4.tsv",
                "qiime2/abundance_tables/abs-abund-table-5.tsv",
                "qiime2/abundance_tables/abs-abund-table-6.tsv",
                "qiime2/abundance_tables/count_table_filter_stats.tsv",
                "qiime2/abundance_tables/feature-table.biom",
                "qiime2/abundance_tables/feature-table.tsv",
                "qiime2/abundance_tables/filtered-table.qza",
                "qiime2/barplot",
                "qiime2/barplot/dist",
                "qiime2/barplot/dist/bundle.js",
                "qiime2/barplot/dist/d3-license.txt",
                "qiime2/barplot/dist/d3-scale-chromatic-license.txt",
                "qiime2/barplot/dist/thenBy-license.txt",
                "qiime2/barplot/dist/vendor.bundle.js",
                "qiime2/barplot/index.html",
                "qiime2/barplot/level-1.csv",
                "qiime2/barplot/level-1.jsonp",
                "qiime2/barplot/level-2.csv",
                "qiime2/barplot/level-2.jsonp",
                "qiime2/barplot/level-3.csv",
                "qiime2/barplot/level-3.jsonp",
                "qiime2/barplot/level-4.csv",
                "qiime2/barplot/level-4.jsonp",
                "qiime2/barplot/level-5.csv",
                "qiime2/barplot/level-5.jsonp",
                "qiime2/barplot/level-6.csv",
                "qiime2/barplot/level-6.jsonp",
                "qiime2/barplot/level-7.csv",
                "qiime2/barplot/level-7.jsonp",
                "qiime2/barplot/q2templateassets",
                "qiime2/barplot/q2templateassets/css",
                "qiime2/barplot/q2templateassets/css/base-template.css",
                "qiime2/barplot/q2templateassets/css/bootstrap.min.css",
                "qiime2/barplot/q2templateassets/css/normalize.css",
                "qiime2/barplot/q2templateassets/css/tab-parent.css",
                "qiime2/barplot/q2templateassets/fonts",
                "qiime2/barplot/q2templateassets/fonts/glyphicons-halflings-regular.eot",
                "qiime2/barplot/q2templateassets/fonts/glyphicons-halflings-regular.svg",
                "qiime2/barplot/q2templateassets/fonts/glyphicons-halflings-regular.ttf",
                "qiime2/barplot/q2templateassets/fonts/glyphicons-halflings-regular.woff",
                "qiime2/barplot/q2templateassets/fonts/glyphicons-halflings-regular.woff2",
                "qiime2/barplot/q2templateassets/img",
                "qiime2/barplot/q2templateassets/img/qiime2-rect-200.png",
                "qiime2/barplot/q2templateassets/js",
                "qiime2/barplot/q2templateassets/js/bootstrap.min.js",
                "qiime2/barplot/q2templateassets/js/child.js",
                "qiime2/barplot/q2templateassets/js/jquery-3.2.0.min.js",
                "qiime2/barplot/q2templateassets/js/parent.js",
                "qiime2/input",
                "qiime2/input/rep-seqs.qza",
                "qiime2/input/table.qza",
                "qiime2/input/taxonomy.qza",
                "qiime2/rel_abundance_tables",
                "qiime2/rel_abundance_tables/rel-table-2.tsv",
                "qiime2/rel_abundance_tables/rel-table-3.tsv",
                "qiime2/rel_abundance_tables/rel-table-4.tsv",
                "qiime2/rel_abundance_tables/rel-table-5.tsv",
                "qiime2/rel_abundance_tables/rel-table-6.tsv",
                "qiime2/rel_abundance_tables/rel-table-ASV.tsv",
                "qiime2/rel_abundance_tables/rel-table-ASV_with-DADA2-tax.tsv",
                "qiime2/representative_sequences",
                "qiime2/representative_sequences/descriptive_stats.tsv",
                "qiime2/representative_sequences/filtered-sequences.qza",
                "qiime2/representative_sequences/rep-seq.fasta",
                "qiime2/representative_sequences/seven_number_summary.tsv",
                "summary_report",
                "summary_report/dada2_taxonomic_classification_per_taxonomy_level.svg",
                "summary_report/rrna_detection_with_barrnap.svg",
                "summary_report/stacked_barchart_of_reads.svg",
                "summary_report/summary_report.html",
                "summary_report/versions.yml"
            ],
            "overall_summary.tsv:md5,cc01727705136cc5d096d373237e1e06",
            "rrna.arc.gff:md5,07229e8767a6165fdc2e392d75aec74a",
            "rrna.bac.gff:md5,5e2dc0da75181e69c61ad49fe877848a",
            "rrna.euk.gff:md5,12cd46457b8fcf5853a6cfba39dd7c63",
            "rrna.mito.gff:md5,737567148a9252d93deb529a06f3239c",
            "ASV_seqs.fasta:md5,f786f2a262c3b7bf1eca475dc2c9bef6",
            "ASV_table.tsv:md5,91ca9b9ab06a873033c29af4f93dfa80",
            "ref_taxonomy.rdp_18.txt:md5,c20d8e186e2893ba7818a0115edb013e",
            "DADA2_stats.tsv:md5,1413d7b7abcdd9344efe78b550112345",
            "DADA2_table.rds:md5,e55f6c929176c853ac4d91123532df9c",
            "DADA2_table.tsv:md5,e9d7e7dfabdd1f9d3bbaa45eba545367",
            "Samplesheet_multi.tsv:md5,25b97db179c42b568e28f00a1c475f78",
            "multiqc_general_stats.txt:md5,5cca03b6090ae25c5c3d4b5f49a7acb8",
            "multiqc_fastqc.txt:md5,7d23984b06e4251f17e1cfce7a04aaae",
            "descriptive_stats.tsv:md5,f6c6527f32005a005fef4d1830874a3e",
            "seven_number_summary.tsv:md5,4b177ed9a865e3320953b69860a435ef"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-12T16:12:59.46389526"
    }
}